tests/ui/empty_enum.rs

```diff
@@ -1,6 +1,7 @@
 #![allow(dead_code)]
 #![warn(clippy::empty_enum)]
-
+// Enable never type to test empty enum lint
+#![feature(never_type)]
```

Comment:
Also nice would be a test `empty_enum_without_never_type.rs` (or a better name) where the feature is not enabled and the `stderr` file is empty aka non-existent.